### DISCUSSION TITLE
Progress of v2?

### DISCUSSION THREAD

**Discussion by tamueller** [basnijholt/adaptive-lighting#288] [General]
Hey,
I've got a question regarding v2: There are some issues with the current version that won't get fixed in v1, as they will be "solved" in v2. Since this is the case since a a few month's now, I have a few questions:
* Is there a rough ETA on v2? If it's around the corner, I'm just gonna wait for it, if it's "when it's done", I'll probably have to invest some time to find workarounds for my issues
* Is it possible to check out the current state of v2? Is it on github?
* Can I somehow help by betatesting and/or contributing to v2? I know a bit of python, but don't have much in depth HA knowledge.

Cheers,
Thomas

**Comment by basnijholt** [comment]
Hi @tamueller,

@RubenKelevra was (is) working on v2, however, I have not seen any of the code yet, so I am not sure whether he has actually found the time to work on it.

**Reply by RubenKelevra** [reply]
Hey @basnijholt,

short update:

Had to do a bit of testing with hardware if my ideas actually work like the feedback loops with a brightness sensor. Which showed some issues with latency - so an off the shelf Aqara sensor is probably suboptimal. But this one I got is not available at the moment anyway. 

So I looked into other sensors and settled on a TCS34725, which is supported by ESPHome. Sadly my first unit was Dead on Arrival. Second unit arrived last week. 

On the code side I did a lot of work on another project in the meantime, to get my feet wet in Custom Components for Home Assistant. I'll start working on this this weekend :)

**Comment by RubenKelevra** [comment]
Hey @tamueller,

here's the roadmap for the next major version, open to votes to change the priority :)

https://github.com/basnijholt/adaptive-lighting/discussions/291

**Comment by daFreeMan** [comment]
Would it be reasonable to ask for regular updates on progress?

Not in an "I demand this be done immediately!" sort of way, but just to give people an idea of what _has_ been accomplished so far. It would also be an opportunity to express appreciation for and congrats on what has been completed so far.
